### PR TITLE
installer: clean up use of Platform type

### PR DESCRIPTION
**Pull request by crawford**
This moves the validation of the platform up to the point of
unmarshalling, reducing the risk that a developer may end up with an
unexpected platform.

### CODE REVIEW

**Comment by enxebre** [comment]
golang test need to be updated to match platform

**Comment by squat** [comment]
Looks like this needs a rebase

File: installer/pkg/config/cluster.go [hunk 1]
```diff
@@ -27,6 +24,27 @@ const (
 	PlatformLibvirt Platform = "libvirt"
 )
 
+// Platform indicates the target platform of the cluster.
+type Platform string
+
+// UnmarshalYAML unmarshals and verifies the platform.
+func (p *Platform) UnmarshalYAML(unmarshal func(interface{}) error) error {
```
[the remark this answers is not in the document]
Comment (crawford) [reply installer/pkg/config/cluster.go]: Yes, but this is a syntactic validation. It shouldn't be possible to instantiate a `Platform` that isn't AWS or Libvirt (though Go doesn't actually let us enforce this).

**Comment by enxebre** [comment]
ok to test

**Pull request merged**